carapace-server/src/main/java/org/carapaceproxy/server/HttpProxyServer.java

```diff
@@ -653,21 +652,24 @@ private void readClusterConfiguration(ConfigurationStore staticConfiguration) th
     private void initGroupMembership() throws ConfigurationNotValidException {
         if (cluster) {
             Map<String, String> peerInfo = new HashMap();
-            String adminHost = adminServerHost;
-            try {
-                adminHost = InetAddress.getLocalHost().getHostName();
-            } catch (UnknownHostException ex) {
-                LOG.log(Level.INFO, "Unable to resolve Admin Server Hostname for peer " + peerId + ". Using " + adminServerHost);
-            }
-            peerInfo.put(PROPERTY_PEER_ADMIN_SERVER_HOST, adminHost);
+            peerInfo.put(PROPERTY_PEER_ADMIN_SERVER_HOST, adminServerHost);          
```

Comment:
this is the point.
I suggest you to introduce a new optional configuration value 
admin.server.advertised.url
this way the user can have full control about the URL to advertise.
You won't need any magic auto discovery